src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,32 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import java.util.concurrent.TimeUnit;
 
 public class AsyncRequestProcessor {
+    private static final int SLEEP_TIME = 200;
     private final Executor executor;
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(() ->
+                cache.computeIfAbsent(userId, this::getUserData), executor);
+    }
+
+    private UserData getUserData(String userId) {
```

Comment:
The method `getUserData` is private and seems to simulate fetching user data with a delay. If this is for demonstration purposes, it should be documented accordingly. If it's for production, you should consider implementing actual data fetching logic or renaming the method to reflect that it's a stub.